progress/reader.go

```diff
@@ -0,0 +1,67 @@
+package progress
+
+import (
+	"crypto/md5"
+	"crypto/sha256"
+	"encoding/hex"
+	"hash"
+	"io"
+	"sync/atomic"
+)
+
+type ReadHasher struct {
+	r      io.Reader
+	md5    hash.Hash
+	sha256 hash.Hash
+}
+
+func NewReadHasher(r io.Reader) *ReadHasher {
```

Comment:
I can't see this being used anywhere, is it still needed?

Reply:
We are calculating the hashes on the studio side for now. Ideally Catalyst would eventually start doing that tho